### PR TITLE
Renamed WorkflowPoller->WorkflowReconciler and can be enabled using property

### PR DESCRIPTION
**Pull request by apanicker-nflx**
Pull Request type
----

- [x] Refactor

Changes in this PR
----
WorkflowPoller is renamed to WorkflowReconciler and is controlled using the property `conductor.workflow-reconciler.enabled`. The property `conductor.workflow-sweeper.enabled` is no longer used.

### CODE REVIEW

File: core/src/main/java/com/netflix/conductor/core/reconciliation/DefaultWorkflowPoller.java [hunk 1]
```diff
@@ -10,42 +10,42 @@
...
-import java.util.concurrent.CompletableFuture;
-
-import static com.netflix.conductor.core.execution.WorkflowExecutor.DECIDER_QUEUE;
-
 @SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection")
 @Component
-@ConditionalOnProperty(name = "conductor.workflow-sweeper.enabled", havingValue = "true", matchIfMissing = true)
-public class WorkflowPoller extends LifecycleAwareComponent {
+@ConditionalOnProperty(name = "conductor.default-workflow-sweeper.enabled", havingValue = "true", matchIfMissing = true)
+public class DefaultWorkflowPoller extends LifecycleAwareComponent implements WorkflowPoller {
```
Comment (aravindanr) [inline core/src/main/java/com/netflix/conductor/core/reconciliation/DefaultWorkflowPoller.java:32]: The `WorkflowPoller` is not called by any of the Conductor internal classes. So, there is not really a need to define a contract that all pollers should follow. Technically, a bespoke poller can be written (without implementing WorkflowPoller) and used just by disabling the `DefaultWorkflowPoller`.

File: core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowRepairService.java [hunk 2]
```diff
@@ -10,12 +10,13 @@
  *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
  *  specific language governing permissions and limitations under the License.
  */
-package com.netflix.conductor.core.execution;
+package com.netflix.conductor.core.reconciliation;
```
Comment (aravindanr) [inline core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowRepairService.java:13]: 👏

File: core/src/main/java/com/netflix/conductor/core/reconciliation/DefaultWorkflowSweeper.java [hunk 3]
```diff
@@ -33,23 +34,23 @@
 
 @SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection")
 @Component
-@ConditionalOnProperty(name = "conductor.workflow-sweeper.enabled", havingValue = "true", matchIfMissing = true)
-public class WorkflowSweeper {
+@ConditionalOnProperty(name = "conductor.default-workflow-sweeper.enabled", havingValue = "true", matchIfMissing = true)
+public class DefaultWorkflowSweeper implements WorkflowSweeper {
```
Comment (aravindanr) [inline core/src/main/java/com/netflix/conductor/core/reconciliation/DefaultWorkflowSweeper.java:38]: Same argument as above. The interface may not be necessary.

File: core/src/main/java/com/netflix/conductor/core/reconciliation/DefaultWorkflowSweeper.java [hunk 4]
```diff
@@ -33,23 +34,23 @@
 
 @SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection")
 @Component
-@ConditionalOnProperty(name = "conductor.workflow-sweeper.enabled", havingValue = "true", matchIfMissing = true)
-public class WorkflowSweeper {
+@ConditionalOnProperty(name = "conductor.default-workflow-sweeper.enabled", havingValue = "true", matchIfMissing = true)
```
Comment (aravindanr) [inline core/src/main/java/com/netflix/conductor/core/reconciliation/DefaultWorkflowSweeper.java:37]: A custom poller could be written that uses the `WorkflowSweeper`. I think the `@Conditional` can be removed.

File: core/src/main/java/com/netflix/conductor/core/reconciliation/DefaultWorkflowPoller.java (hunk 1 above)
Comment (apanicker-nflx) [reply core/src/main/java/com/netflix/conductor/core/reconciliation/DefaultWorkflowPoller.java]: good point, changed

File: core/src/main/java/com/netflix/conductor/core/reconciliation/DefaultWorkflowSweeper.java (hunk 4 above)
Comment (apanicker-nflx) [reply core/src/main/java/com/netflix/conductor/core/reconciliation/DefaultWorkflowSweeper.java]: removed

File: core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java [hunk 5]
```diff
@@ -10,30 +10,27 @@
...
 
-import java.util.Optional;
-import java.util.concurrent.CompletableFuture;
-
-import static com.netflix.conductor.core.config.SchedulerConfiguration.SWEEPER_EXECUTOR_NAME;
-import static com.netflix.conductor.core.execution.WorkflowExecutor.DECIDER_QUEUE;
-
 @SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection")
 @Component
-@ConditionalOnProperty(name = "conductor.workflow-sweeper.enabled", havingValue = "true", matchIfMissing = true)
```
Comment (aravindanr) [inline core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java:36]: `com.netflix.conductor.test.base.AbstractSpecification` no longer needs `Optional<WorkflowSweeper>`

File: core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java [hunk 6]
```diff
@@ -46,10 +43,8 @@
     private static final String CLASS_NAME = WorkflowSweeper.class.getSimpleName();
 
     @Autowired
-    public WorkflowSweeper(WorkflowExecutor workflowExecutor,
-                           Optional<WorkflowRepairService> workflowRepairService,
-                           ConductorProperties properties,
-                           QueueDAO queueDAO) {
+    public WorkflowSweeper(WorkflowExecutor workflowExecutor, Optional<WorkflowRepairService> workflowRepairService,
```
Comment (aravindanr) [inline core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java:46]: The formatting in many places seem off. Maybe try reformatting in the IDE before commit?

**Review by aravindanr — APPROVED** [review]

File: core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java (hunk 5 above)
Comment (apanicker-nflx) [reply core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java:36]: changed

File: core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java (hunk 6 above)
Comment (apanicker-nflx) [reply core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java:46]: tried re-formatting but to no effect, will fix in a later pr

**Pull request merged**